### PR TITLE
Add VS2022 support and misc

### PR DESCRIPTION
**Pull request by namrog84**
* Add VS2022 support
* Suppress AddLogListener NativeEventListener warnings.  
    *  "Warning: `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.  
    *  Also warning for removeListeners method
* Remove unnecessary reference dependency

### CODE REVIEW

File: windows/OrientationWindows/OrientationWindows.vcxproj [hunk 1]
```diff
@@ -155,11 +156,6 @@
   <ItemGroup>
     <Midl Include="ReactPackageProvider.idl" />
   </ItemGroup>
-  <ItemGroup>
-    <ProjectReference Include="$(ReactNativeWindowsDir)\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj">
```
Comment (namrog84) [inline windows/OrientationWindows/OrientationWindows.vcxproj:159]: I believe this was only originally needed before 
Microsoft.ReactNative.Uwp.CppLib.props and Microsoft.ReactNative.Uwp.CppLib.targets was added. And those were added long ago.

**Comment by namrog84** [comment]
Hi @wonday, is there anything I can do or change that would help get this in? 👍
Thanks!

**Review by Nesh108 — APPROVED** [review]

**Comment by Nesh108** [comment]
@wonday we definitely need this PR otherwise this package won't work with newer versions of React-Native.

**Comment by wonday** [comment]
Thank you.

**Pull request merged**